storage/tokudb/ha_tokupart.cc

```diff
@@ -0,0 +1,83 @@
+
+handler *ha_tokupart::get_file_handler(TABLE_SHARE *share,
+                                       MEM_ROOT *alloc,
+                                       handlerton *db_type) {
+  DBUG_ASSERT(db_type == tokudb_hton);
+  ha_tokudb *file = new(alloc) ha_tokudb(db_type, share);
+  file->init();
+  return file;
+}
+
+
+/**
+  Clone the open and locked partitioning handler.
+
+  @param  mem_root  MEM_ROOT to use.
+
+  @return Pointer to the successfully created clone or NULL
+
+  @details
+  This function creates a new Partition_base handler as a clone/copy. The
+  original (this) must already be opened and locked. The clone will use
+  the originals m_part_info.
+  It also allocates memory for ref + ref_dup.
+  In Partition_base::open() it will clone its original handlers partitions
+  which will allocate then on the correct MEM_ROOT and also open them.
+*/
+
+handler *ha_tokupart::clone(const char *name, MEM_ROOT *mem_root)
+{
+  ha_tokupart *new_handler;
+
+  DBUG_ENTER("Partition_base::clone");
+
+  /* If this->table == NULL, then the current handler has been created but not
+  opened. Prohibit cloning such handler. */
+  if (!table)
+    DBUG_RETURN(NULL);
+
+  new_handler= new (mem_root) ha_tokupart(ht, table_share, m_part_info,
+                                           this, mem_root);
+  if (!new_handler)
+    DBUG_RETURN(NULL);
```

Comment:
Same, this seems like an OOM problem so maybe a hard assertion/abort is in order?